osutil/disks/luks.go

```diff
@@ -39,8 +39,8 @@ var (
 	luksUUIDPatternRe = regexp.MustCompile(`^CRYPT-LUKS2-([0-9a-f]{32})$`)
 )
 
-func cryptLuks2DeviceMapperBackResolver(dmUUID, dmName []byte) (dev string, ok bool) {
-	if !strings.HasPrefix(string(dmUUID), "CRYPT-LUKS") {
+func cryptLuks2DeviceMapperBackResolver(opts *deviceMapperBackResolversOpts) (dev string, ok bool) {
+	if !strings.HasPrefix(string(opts.dmUUID), "CRYPT-LUKS") {
```

Comment:
This might crash if called with opts being nil due to a refactor that isn't obvious.

Let's add a check that opts is not nil, or pass opts by value (with a string and two slices it's probably okay).

Reply:
Added.